pkg/gossip/infostore.go

```diff
@@ -360,12 +374,7 @@ func (is *infoStore) combine(
 	localNodeID := is.nodeID.Get()
 	for key, i := range infos {
 		if i.NodeID == localNodeID {
-			// Ignore infos received from the remote node which originated at the
-			// local node. These infos will usually be ignored by addInfo due to
-			// having a hops value that is too large, but during startup we might
-			// inadvertently accept one of these infos until the local node has
-			// gossiped the same info.
-			continue
+			ratchetMonotonic(i.OrigStamp)
```

Comment:
So in this scenario it's possible for a node to restart, gossip some new info, but have that new info effectively ignored everywhere because its timestamp is lower than what it gossiped before the restart. I suppose that's better than the local node having a different gossip state than remote nodes, but it'd be nice if it triggered a re-gossip of local information. It's certainly an edge case, and an existing one at that, but having info from the previous instantiation of the node prioritized over the new info could be an issue, particularly if the node had a different address before the restart.

Might be worth filing an issue about, although priority-wise I wouldn't expect us to act on it very soon.

Reply:
> So in this scenario it's possible for a node to restart, gossip some new info, but have that new info effectively ignored everywhere because its timestamp is lower than what it gossiped before the restart. I suppose that's better than the local node having a different gossip state than remote nodes, but it'd be nice if it triggered a re-gossip of local information.

Yes. To be clear, that scenario was possible before this PR. A backward clock jump will cause a crash and the new process will gossip values that will be ignored by the rest of the cluster until its clock advances. I'll file an issue about this. I think we should fix it for 2.2